astropy/tests/helper.py

```diff
@@ -304,14 +304,27 @@ def _save_coverage(cov, result, rootdir, testing_path):
     # Python 2.x.
     if six.PY2:
         d = cov.data
-        cov._harvest_data()
-        for key in d.lines.keys():
-            new_path = os.path.relpath(
-                os.path.realpath(key),
-                os.path.realpath(testing_path))
-            new_path = os.path.abspath(
-                os.path.join(rootdir, new_path))
-            d.lines[new_path] = d.lines.pop(key)
+        try:
+            # Coverage 4.0: _harvest_data has been renamed to get_data, the
+            # lines dict is private
+            cov.get_data()
+            for key in d._lines.keys():
+                new_path = os.path.relpath(
+                    os.path.realpath(key),
+                    os.path.realpath(testing_path))
+                new_path = os.path.abspath(
+                    os.path.join(rootdir, new_path))
+                d._lines[new_path] = d._lines.pop(key)
+        except AttributeError:
+            # Coverage < 4.0
+            cov._harvest_data()
+            for key in d.lines.keys():
+                new_path = os.path.relpath(
+                    os.path.realpath(key),
+                    os.path.realpath(testing_path))
+                new_path = os.path.abspath(
+                    os.path.join(rootdir, new_path))
+                d.lines[new_path] = d.lines.pop(key)
```

Comment:
Yes, I did this in quick and dirty mode to know if the issue was limited to this code block. I will probably not have time to clean this before tomorrow si feel free to integrate a better patch.